vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/ZephyrFacade.java

```diff
@@ -0,0 +1,130 @@
+    private final ZephyrConfiguration zephyrConfiguration;
+
+    public ZephyrFacade(JiraFacade jiraFacade, JiraClient client, ZephyrConfiguration zephyrConfiguration)
+    {
+        this.jiraFacade = jiraFacade;
+        this.client = client;
+        this.zephyrConfiguration = zephyrConfiguration;
+    }
+
+    @Override
+    public Integer createExecution(String execution) throws IOException
+    {
+        String responseBody = client.executePost(CREATE_EXECUTION_ENDPOINT, execution);
+        List<Integer> executionId = JsonPathUtils.getData(responseBody, EXECUTION_ID_JSON_PATH);
+        return executionId.get(0);
+    }
+
+    @Override
+    public void updateExecutionStatus(int executionId, String executionBody) throws IOException
+    {
+        client.executePut(String.format(UPDATE_EXECUTION_STATUS_ENDPOINT, executionId), executionBody);
+    }
+
+    @Override
+    public ZephyrConfiguration prepareConfiguration() throws IOException
+    {
+        notBlank(zephyrConfiguration.getProjectKey(), "Property 'zephyr.project-key=' should not be empty");
+        notBlank(zephyrConfiguration.getVersionName(), "Property 'zephyr.version-name=' should not be empty");
+        notBlank(zephyrConfiguration.getCycleName(), "Property 'zephyr.cycle-name=' should not be empty");
+        notBlank(zephyrConfiguration.getFolderName(), "Property 'zephyr.folder-name=' should not be empty");
+
+        Project project = jiraFacade.getProject(zephyrConfiguration.getProjectKey());
+        String projectId = project.getId();
+        zephyrConfiguration.setProjectId(projectId);
+
+        String versionId = findVersionId(project);
+        zephyrConfiguration.setVersionId(versionId);
+
+        String projectAndVersionUrlQuery = String.format("projectId=%s&versionId=%s", projectId, versionId);
+
+        String cycleId = findCycleId(projectAndVersionUrlQuery);
+        zephyrConfiguration.setCycleId(cycleId);
+
+        String folderId = findFolderId(cycleId, projectAndVersionUrlQuery);
+        zephyrConfiguration.setFolderId(folderId);
+        return zephyrConfiguration;
+    }
+
+    private String findVersionId(Project project)
+    {
+        Optional<Version> version = project.getVersions().stream().filter(
+            v -> zephyrConfiguration.getVersionName().equals(v.getName())).findFirst();
+        isTrue(version.isPresent(), "Version with name '%s' does not exist", zephyrConfiguration.getVersionName());
+        return version.get().getId();
+    }
+
+    private String findCycleId(String projectAndVersionUrlQuery) throws IOException
+    {
+        String json = client.executeGet(ZAPI_ENDPOINT + "cycle?" + projectAndVersionUrlQuery);
+        Map<String, Map<String, String>> cycles = JsonPathUtils.getData(json, "$");
+        cycles.remove("recordsCount");
+        Iterator<Map.Entry<String, Map<String, String>>> itr = cycles.entrySet().iterator();
+        String cycleId = "";
+        while (itr.hasNext())
```

Comment:
do you really need a loop here? can't you use json path to find the needed value?

Reply:
I didn't find ability to write json path which extract parent node by condition for child.
For example: by `name` need to get id.
`{
   "7502":{
      "versionName":"Release 1",
      "projectKey":"TEST",
      "name":"Sprint 1"
   },
   "7592":{
      "versionName":"Release 2",
      "projectKey":"TEST",
      "name":"Sprint 2"
   },
   "recordsCount":2
}`

Reply:
that's tru, there is no way to get keys using only JSON path